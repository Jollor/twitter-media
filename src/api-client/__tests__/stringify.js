const path = require('path');
const fs = require('fs');
const stringify = require('../stringify');

const image = fs.readFileSync(path.join(__dirname, '..', '..', '..', 'assets', 'image.jpg'));

test('stringify an object containing a buffer', () => {
    const actual = stringify({
        media: image
    });

    expect(actual).toMatchSnapshot();
});

test('stringify an object containing a buffer in a nested value', () => {
    const actual = stringify({
        method: 'GET',
        formData: {
            media: image,
            media_id: '123'
        }
    });

    expect(actual).toMatchSnapshot();
});
