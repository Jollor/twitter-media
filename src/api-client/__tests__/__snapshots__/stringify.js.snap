// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringify an object containing a buffer 1`] = `"{\\"media\\":\\"<Buffer ff d8 ff e0 0 ...>\\"}"`;

exports[`stringify an object containing a buffer in a nested value 1`] = `"{\\"method\\":\\"GET\\",\\"formData\\":{\\"media\\":\\"<Buffer ff d8 ff e0 0 ...>\\",\\"media_id\\":\\"123\\"}}"`;
